File: Data Structure/Hash/Easy/LET_no409_Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        if not s:
            return 0

        res, odds = 0, 0
        for num in collections.Counter(s).values():
            res += num
            if num % 2:
                odds += 1
        if odds:
            res -= odds - 1
        
        return res

File: Data Structure/Hash/Easy/LET_no409_Longest Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> counter;
        for (char c : s) 
            counter[c]++;

        int res = 0, odds = 0;
        for (auto it = counter.begin(); it != counter.end(); it++) {
            odds += it->second % 2;
            res += it->second;
        }
        if (odds > 0)
            res -= odds - 1;
        
        return res;
    }
};